src/main/java/ca/unknown/bot/use_cases/TimerInteractor.java

```diff
@@ -22,15 +27,24 @@ public class TimerInteractor extends ListenerAdapter {
 
     @Override
     public void onSlashCommandInteraction(SlashCommandInteractionEvent event) {
-        // !preset {worktime} {breaktime} {iteration} {name}
-        if (event.getName().equals("timer_create")) {
+        // /timer_create {workTime} {breakTime} {iteration} {name}
+        if (event.getName().equals("timer_create")) { // Slash command for creating a timer
             try {
+                User user = event.getUser();
```

Comment:
I think that the parts where you get input from the user (ie. name, workTime, breakTime, interaction) should be the only things in your controller and the class should be used to create a new input data entity. and then in line 32 you call the pomodoro factory to make your object with create(inputdata.getWorkTime(), inputdata.getBreakTime(), inputdata.getIteration(), inputdata.getName()) to make the timer object. then, call the DAO to save it (which you were previously doing in your controller object). your ideas and syntax are right, but just that the controller should only be used for turning the user data into data the interactor needs and accessing that data from an input data class with get methods. saving should be a method in the DAO and also checking for duplicates is in the DAO as well i think. please let me know what you guys think about this line of reasoning!

Reply:
I am really thankful for your considerate feedback. It gave me an opportunity to rethink about the CA and the structure of how the bot saves a new timer. 

I have designed TimerInteractor to be the class that directly gets inputs from an instance of SlashCommandInteractionEvent (which is /timer_create in this case) since it is the one that gets such an event instance. But, instead of doing something with them, it delivers the inputs to TimerController, and TimerController converts these into a map, which is like a default structure GSON returns when you ask it to convert a json file into a java object, so that TimerDAO finds it easy to compare with what's in timer_repository. I had to leave TimerDAO up to decide how the map from TimerController should be read as timer_repository can dynamically change. 

I changed the name of the method TimerInteractor calls when processing inputs (savePomodoro --> convertTimerInput) to avoid confusion. (My bad! I suck at naming stuff). Also, I decided not to use PomodoroFactory in TimerController cuz it doesn't have any subtypes. I agree with your idea that TimerController doesn't need a method for checking duplicates as TimerDAO already has it. I will make TimerInteractor directly interacts with TimerDAO to do so, but this will cause TimerInteractor to convert the user into its String representation in it.